### PR TITLE
Set all node conditions to Unknown when node is unreachable

### LINKED ISSUE #36273: When node is unreachable, master should show all node conditions as Unknown
I just found a cluster like this
```
Conditions:
  Type			Status		LastHeartbeatTime			LastTransitionTime			Reason				Message
  ----			------		-----------------			------------------			------				-------
  NetworkUnavailable 	False 		Sat, 05 Nov 2016 00:01:27 +0000 	Sat, 05 Nov 2016 00:01:27 +0000 	RouteCreated 			RouteController created a route
  OutOfDisk 		Unknown 	Thu, 03 Nov 2016 12:18:28 +0000 	Thu, 03 Nov 2016 12:19:12 +0000 	NodeStatusUnknown 		Kubelet stopped posting node status.
  MemoryPressure 	False 		Thu, 03 Nov 2016 12:18:28 +0000 	Thu, 03 Nov 2016 12:05:44 +0000 	KubeletHasSufficientMemory 	kubelet has sufficient memory available
  DiskPressure 		False 		Thu, 03 Nov 2016 12:18:28 +0000 	Thu, 03 Nov 2016 12:05:44 +0000 	KubeletHasNoDiskPressure 	kubelet has no disk pressure
  Ready 		Unknown 	Thu, 03 Nov 2016 12:18:28 +0000 	Thu, 03 Nov 2016 12:19:12 +0000 	NodeStatusUnknown 		Kubelet stopped posting node status.
```

cc/ @dchen1107

### PR DESCRIPTION
**Pull request by andrewsykim**
**What this PR does / why we need it**:
Sets all node conditions to Unknown when node does not report status/unreachable

**Which issue this PR fixes** 
fixes https://github.com/kubernetes/kubernetes/issues/36273

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36592)

### CODE REVIEW

**Comment by davidopp** [comment]
@k8s-bot ok to test

**Comment by davidopp** [comment]
I think this is a useful PR. Sorry it fell off my radar and won't make it into 1.5....

**Comment by andrewsykim** [comment]
@davidopp no worries! It seems like the list of node conditions may have changed, wondering what the best way of keeping the node conditions in sync would be.

**Comment by davidopp** [comment]
@gmarek is there a chance you can review this? I think the current behavior is confusing.

File: pkg/controller/node/nodecontroller.go [hunk 1]
```diff
@@ -740,35 +740,37 @@ func (nc *NodeController) tryUpdateNodeStatus(node *v1.Node) (time.Duration, v1.
...
-		} else {
-			glog.V(4).Infof("node %v hasn't been updated for %+v. Last out of disk condition is: %+v",
-				node.Name, nc.now().Time.Sub(savedNodeStatus.probeTimestamp.Time), oodCondition)
-			if oodCondition.Status != v1.ConditionUnknown {
-				oodCondition.Status = v1.ConditionUnknown
-				oodCondition.Reason = "NodeStatusUnknown"
-				oodCondition.Message = fmt.Sprintf("Kubelet stopped posting node status.")
-				oodCondition.LastTransitionTime = nc.now()
+		// remaining node conditions should also be set to Unknown
+		remainingNodeConditionTypes := []v1.NodeConditionType{v1.NodeOutOfDisk, v1.NodeMemoryPressure, v1.NodeDiskPressure, v1.NodeNetworkUnavailable}
```
Comment (gmarek) [inline pkg/controller/node/nodecontroller.go]: I'm not sure about NetworkUnavailable - IIRC this is basically 'route is created' condition, so Kubelet isn't involved in setting this. @wojtek-t 

Rest seems fine.

File: pkg/controller/node/nodecontroller.go [hunk 2]
```diff
@@ -740,35 +740,37 @@ func (nc *NodeController) tryUpdateNodeStatus(node *v1.Node) (time.Duration, v1.
...
+					LastTransitionTime: nc.now(),
+				})
+			} else {
+				glog.V(4).Infof("node %v hasn't been updated for %+v. Last %v is: %+v",
+					node.Name, nc.now().Time.Sub(savedNodeStatus.probeTimestamp.Time), nodeConditionType, currentCondition)
+				if currentCondition.Status != v1.ConditionUnknown {
+					currentCondition.Status = v1.ConditionUnknown
+					currentCondition.Reason = "NodeStatusUnknown"
+					currentCondition.Message = "Kubelet stopped posting node status."
+					currentCondition.LastTransitionTime = nc.now()
```
Comment (gmarek) [inline pkg/controller/node/nodecontroller.go]: We probably want to have the same timestamp for all condition, so please grab 'now' before the loop.

**Comment by andrewsykim** [comment]
@gmarek address comments, some tests have failed and seem unrelated to my changes.

**Comment by gmarek** [comment]
@k8s-bot gci gce e2e test this

**Comment by gmarek** [comment]
@k8s-bot gci gke smoke e2e test this

**Comment by gmarek** [comment]
@k8s-bot gci gke e2e test this

**Comment by gmarek** [comment]
Sorry - it completely fell of my radar.

**Comment by gmarek** [comment]
@k8s-bot gce e2e test this.

**Pull request merged**